src/inference/src/dev/core_impl.cpp

```diff
@@ -391,25 +386,26 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::string& mod
     auto plugin = get_plugin(parsed._deviceName);
     ov::SoPtr<ov::ICompiledModel> res;
 
-    auto cacheManager =
-        coreConfig.get_cache_config_for_device(parsed._deviceName, device_supports_cache_dir(plugin), parsed._config)
-            ._cacheManager;
+    property_manager.merge_property(parsed._config, plugin.get_name());
```

Comment:
`merge_property` changes the global Core state, while we are not allowed to do it from `compile_model`, `query_model` and `import_model`.
Only `set_property` can change Core state and how it handles specific devices.

Reply:
So, we will not support set core property by `compile_model()`, `query_model() `and `import_model()`?
It will cause some behavior changes, maybe impact customer application and many CI test cases>

Reply:
> we will not support set core property by compile_model(), query_model() and import_model()?

We will support, but they should affect only current call of `compile_model`, etc:
```c++
ov::Core core;
core.compile_model("model1", "CPU", ov::cache_dir("/tmp"); // here you change global state
core.compile_model("model2", "CPU"); // while user does not want second compile_model's call to do caching
```

Reply:
Is there any actual requirement that set different cache dirs for different models with the same plugin?
If yes, I think we need go back the previous solution: pass core properties to compile_model() and plugin decide if use or ignore/remove it.

Reply:
CC @ilyachur

Reply:
I guess if we keep core_properties takes effect only for each `compile_model()` calling, we should pass it into plugin rather than cache it to let plugin to fetch it from core.

Supposed ov::cache_dir will be removed from plugin's property, it is ok, but for other core_properties which needed in plugin, it is hard to be applied for each `compiled_model()` calling due to muti-models in mutil-threads use cases.

So, my idea about core_properties is:
  1). core_properties will be cached if user call `core.set_property()`
  2). core_properties will not be cached if user call `core.compile_model()` `core.import_model` `core.query_model`
         (1). `ov::cache_dir` will be applied directly
         (2). `ov::auto_batch_timeout` and `ov::hint::allow_auto_batching` `ov::force_tbb_terminate` will be removed or not allowed set.